org.eclipse.scanning.api/src/org/eclipse/scanning/api/points/AbstractGenerator.java

```diff
@@ -217,6 +216,7 @@ public int getRank() throws GeneratorException {
 	/**
 	 * Please override this method, the default creates all points and
 	 * returns their size
+	 * @throws GeneratorException
 	 */
 	protected int sizeOfValidModel() throws GeneratorException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'org.eclipse.scanning.api.points.GeneratorException', as it cannot be thrown from method's body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)